File: VB/ClassLibrary2/PersistentClass2.vb
Imports DevExpress.Xpo
Imports DevExpress.Persistent.Base

Namespace ClassLibrary2

    <DefaultClassOptions>
    Public Class PersistentClass2
        Inherits XPObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Private _PersistentPropertyX As String

        Public Property PersistentPropertyX As String
            Get
                Return _PersistentPropertyX
            End Get

            Set(ByVal value As String)
                SetPropertyValue(Of String)(NameOf(PersistentClass2.PersistentPropertyX), _PersistentPropertyX, value)
            End Set
        End Property
    End Class
End Namespace

File: CS/ClassLibrary2/PersistentClass2.cs
using System;
using System.Linq;
using DevExpress.Xpo;
using System.Collections.Generic;
using DevExpress.Persistent.Base;

namespace ClassLibrary2 {
    [DefaultClassOptions]
    public class PersistentClass2 : XPObject {
        public PersistentClass2(Session session) : base(session) { }
        private string _PersistentPropertyX;
        public string PersistentPropertyX {
            get { return _PersistentPropertyX; }
            set { SetPropertyValue<string>(nameof(PersistentPropertyX), ref _PersistentPropertyX, value); }
        }
    }
}
